packages/exhibitor-migration-script/extra/exhibitor-migration-script/bin/dcos-exhibitor-migrate-perform

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+import copy
+
+import requests
+import sys
+from requests import RequestException
+
+ZOOKEEPER_DATA_DIR_START = '/var/lib/zookeeper/snapshot'
+ZOOKEEPER_LOG_DIR_START = '/var/lib/zookeeper/transactions'
+LOG_INDEX_DIR_START = '/var/lib/zookeeper/transactions'
+
+ZOOKEEPER_DATA_DIR_GOAL = '/var/lib/dcos/exhibitor/zookeeper/snapshot'
+ZOOKEEPER_LOG_DIR_GOAL = '/var/lib/dcos/exhibitor/zookeeper/transactions'
+LOG_INDEX_DIR_GOAL = '/var/lib/dcos/exhibitor/zookeeper/transactions'
+
+
+class ValidationError(Exception):
+    def __init__(self, msg, exit_status=8):
+        self.msg = msg
+        self.exit_status = exit_status
+
+
+def get_config_json(response):
+    if response.status_code == 200 and response.headers['Content-Type'] == 'application/json':
+        return response.json()['config']
+    else:
+        raise ValidationError("Error reading configuration from Exhibitor", 2)
+
+
+def migration_already_complete(config):
+    if config['rollInProgress'] is True:
+        return False
+    else:
+        return (config['zookeeperDataDirectory'] == ZOOKEEPER_DATA_DIR_GOAL and
+                config['zookeeperLogDirectory'] == ZOOKEEPER_LOG_DIR_GOAL and
+                config['logIndexDirectory'] == LOG_INDEX_DIR_GOAL)
+
+
+def assert_can_migrate(config):
+    if config['rollInProgress'] is True:
+        raise ValidationError("Rolling upgrade in progress. {}".format(config['rollStatus']), 4)
+    else:
+        zk_data_directory = config['zookeeperDataDirectory']
```

Comment:
nit: might make sense to have a helper function for these `validate_config_key('zookeperDataDirectory', 'Zookeeper Data Directory', ZOOKEEPER_DATA_DIR_START)` helper for these to reduce repeated code